lib/exportModules.nix

```diff
@@ -21,35 +30,20 @@ let
 
       **/
 
-    let
-
-      removeSuffix = suffix: str:
-        let
-          sufLen = builtins.stringLength suffix;
-          sLen = builtins.stringLength str;
-        in
-        if sufLen <= sLen && suffix == builtins.substring (sLen - sufLen) sufLen str then
-          builtins.substring 0 (sLen - sufLen) str
-        else
-          str;
-
-      genAttrs' = func: values: builtins.listToAttrs (map func values);
-
-      hasFileAttr = o: builtins.hasAttr "_file" o;
-      peek = f: f (builtins.functionArgs f);
-
-    in
-
     genAttrs'
       (arg:
 
-        # a regular path to be imported
-        if builtins.isPath arg then
+        # a regular file to be imported
+        if pathIsRegularFile arg then
```

Comment:
Unfotunately the implementatino does not work with `arg` being a (module) function, which is perfectly legal, as lon as it exhibits a `_file` attribute.

I'll push a fix.